src/main/scala/com/vngrs/etl/Extractor.scala

```diff
@@ -0,0 +1,27 @@
+package com.vngrs.etl
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+import scala.reflect.ClassTag
+
+trait Extractor[A] {
+  protected[etl] def apply(sc: SparkContext): RDD[A]
```

Comment:
I just tried and apparently see that Scala doesn't let us do that without a hack like this

```
trait CustomExtractor[A] extends Extractor[A] {

    protected def extract(sc: SparkContext): RDD[A]
    protected[etl] def apply(sc: SparkContext): RDD[A] = extract(sc)

  }

val customerExtractor = new CustomExtractor[String] {
  protected def extract(sc: SparkContext): RDD[A] = ???
}
```